terraform/aptos-node/aws/kubernetes.tf

```diff
@@ -94,6 +94,21 @@ resource "helm_release" "calico" {
 }
 
 locals {
+  # for performance reasons, require validators and fullnodes to have their own k8s worker nodes on scheduling
+  validator_and_fullnode_affinity = {
+    podAntiAffinity = {
+      requiredDuringSchedulingIgnoredDuringExecution = [{
+        labelSelector = {
+          matchExpressions = [{
+            key      = "app.kubernetes.io/name"
+            operator = "In"
+            values   = ["validator", "fullnode"]
+          }]
+        }
+        topologyKey = "kubernetes.io/hostname"
```

Comment:
Yea we're making that assumption. In the default deployment config, that's the case https://github.com/aptos-labs/aptos-core/blob/main/terraform/helm/aptos-node/values.yaml#L19-L24 

If operators choose to change their instance types, they must also scale up the CPU/mem allocation of their validator/fullnode pods

Reply:
I guess we can also just soft-enforce the antiaffinity by assuming operators will increasing the CPU/mem allocation in correlation with instance size...

Reply:
Actually I'll close this for now... I think we can just rely on operators using higher resource allocation for their pods rather than potentially wastefully forcing this antiaffinity.